macos/daemon/helper.sh

```diff
@@ -216,7 +216,7 @@ collect_new_service_dns() {
 		[[ $service == "*"* ]] && service="${service:1}"
 		found_services="$found_services;$service"
 		array_contains "$SERVICE_DNS" "$service" && continue
-		get_response="$(cmd networksetup -getdnsservers "$service")"
+		get_response="$(cmd networksetup -getdnsservers "$service" |tr '\n' '-')"
```

Comment:
these changes seem unrelated to `cmd_recover`, and my bash isn't good enough to just know what they're doing. Can you elaborate?

Reply:
Right! But this fixes a related DNS issue. The issue was that, if the device, before enabling the VPN, has 2 DNS entries, we were treated them as "Empty" because of this: https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/macos/daemon/helper.sh#L223
Let's say that this is related bug fix.